src/autoscaler/eventgenerator/aggregator/aggregator.go

```diff
@@ -80,3 +89,24 @@ func (a *Aggregator) startWork() {
 		}
 	}
 }
+
+func (a *Aggregator) startSavingAppMetric() {
+	appMetricArray := []*models.AppMetric{}
+	ticker := a.cclock.NewTicker(a.saveInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-a.doneChan:
+			return
+		case appMetric := <-a.appMetricChan:
+			appMetricArray = append(appMetricArray, appMetric)
+		case <-ticker.C():
+			go func(appMetricDB db.AppMetricDB, metrics []*models.AppMetric) {
+				appMetricDB.SaveAppMetricsInBulk(metrics)
+				metrics = nil
```

Comment:
why set metrics to be nil?

Reply:
@boyang9527  After "metrics" has been saved to db, it should be set to nil so that gc can work.

Reply:
metrics is passed in with appMetricArray. You don't need to set to nil